extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java

```diff
@@ -558,28 +559,38 @@ private void micrometerJvmGcMetrics(MetricRegistry registry, ShutdownContext shu
         if (!ImageInfo.inImageCode()) {
             MicrometerGCMetrics gcMetrics = new MicrometerGCMetrics();
 
-            registry.register(new ExtendedMetadata("jvm.gc.max.data.size",
-                    MetricType.GAUGE,
-                    MetricUnits.BYTES,
-                    "Max size of old generation memory pool",
-                    true), new LambdaGauge(gcMetrics::getMaxDataSize));
-            registry.register(new ExtendedMetadata("jvm.gc.live.data.size",
-                    MetricType.GAUGE,
-                    MetricUnits.BYTES,
-                    "Size of old generation memory pool after a full GC",
-                    true), new LambdaGauge(gcMetrics::getLiveDataSize));
-            registry.register(new ExtendedMetadata("jvm.gc.memory.promoted",
-                    MetricType.COUNTER,
-                    MetricUnits.BYTES,
-                    "Count of positive increases in the size of the old generation memory pool before GC to after GC",
-                    true,
-                    "jvm_gc_memory_promoted_bytes_total"), new LambdaCounter(gcMetrics::getPromotedBytes));
-            registry.register(new ExtendedMetadata("jvm.gc.memory.allocated",
-                    MetricType.COUNTER,
-                    MetricUnits.BYTES,
-                    "Incremented for an increase in the size of the young generation memory pool after one GC to before the next",
-                    true,
-                    "jvm_gc_memory_allocated_bytes_total"), new LambdaCounter(gcMetrics::getAllocatedBytes));
+            registry.register(new ExtendedMetadataBuilder()
+                    .withName("jvm.gc.max.data.size")
+                    .withType(MetricType.GAUGE)
+                    .withUnit(MetricUnits.BYTES)
+                    .withDescription("Max size of old generation memory pool")
+                    .skipsScopeInOpenMetricsExportCompletely(true)
+                    .build(), new LambdaGauge(gcMetrics::getMaxDataSize));
```

Comment:
While we are at it, can we remove the use of lambdas in runtime code please?

Reply:
Do we really have to do that? 
see https://github.com/quarkusio/quarkus/pull/3732#issuecomment-526520841 
I'm not an expert on this topic but it seems the memory penalty is negligible

Reply:
We don't **have** to do it, but it's been a general guideline for the runtime code of the project that we generally do try to follow.

Reply:
Ok I'll do that then, but give me some time, hopefully will get round to it tomorrow :)

Reply:
Sure thing, no rush :)